### DISCUSSION TITLE
Running Typescript file from command line in NextJS

### DISCUSSION THREAD

**Discussion by MikeCarbone** [vercel/next.js#37901] [Help]
Hello!

I'm trying to run a typescript file from the command line that runs through another typescript file and tests some functions. In the end I want to run `npm run typescriptFile.ts`. Despite my many dives into `ts-node` and trying to run `esrun` (in various setups), I'm having no luck.

As an example, the code could be something as simple as this:
```js
import type Course from 'interfaces/Course'

const course : Course = { test: "object" }
console.log(course);
```
And I can't figure out the right configuration to make it work!

Some of the various errors I'm running into:
- Errors importing local modules
- Running into path errors (`interfaces/Course` is configured in `tsconfig.json`, but doesn't work unless I use relative paths) 
- Should I be ending the import with `.js`?
- Sometimes running into module import errors `Module not found`
- Some people say I need `"type": "module"` in my `package.json`, some say this is causing issues

I've tried so many different combinations of approaches, I've been reading StackOverflow threads, trying and trying without luck. Always an error. `ts-node` comes up with a different error no matter how I spin it.

If somebody could provide an example repository that accomplishes this task so I can see the configuration, I'd be very grateful.

Let me know if you have questions.

**Comment by msafi** [comment]
I think for this to work, the TypeScript file would need to be compiled and executed by Webpack, using Next.js's Webpack configurations. I don't know enough about Next.js and webpack to do this. Hopefully someone more knowledgeable can help!

**Comment by falsepopsky** [comment]
I use this `ts-node --compiler-options {\"module\":\"CommonJS\"} typescriptFile.ts`
